apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexService.java

```diff
@@ -17,10 +20,38 @@
 public class DexService {
     private static final Logger LOG = LoggerFactory.getLogger(DexService.class);
 
-    private EthereumWalletService ethereumWalletService = CDI.current().select(EthereumWalletService.class).get();
+    private EthereumWalletService ethereumWalletService;
+    private DexOfferDao dexOfferDao;
 
-    public ExchangeBalances getBalances(String ethAddress, String paxAddress){
+    @Inject
+    public DexService(EthereumWalletService ethereumWalletService, DexOfferDao dexOfferDao) {
+        this.ethereumWalletService = ethereumWalletService;
+        this.dexOfferDao = dexOfferDao;
+    }
 
+    public void updateOffer (DexOffer offer){
+        dexOfferDao.save(offer);
+    }
+
+    public DexOffer getOfferByTransactionId(Long transactionId){
+        return dexOfferDao.getByTransactionId(transactionId);
+    }
+
+    public void deleteOfferByTransactionId(Long transactionId){
+        dexOfferDao.deleteByTransactionId(transactionId);
+    }
+
+    public void saveOffer (DexOffer offer){
+        if(dexOfferDao.getByTransactionId(offer.getTransactionId()) == null){
```

Comment:
Annotate each jdbi dao method call with @Transactional annotation to avoid connection leaks